<launch>

    <arg name="robot_name"            default="ente" />
    <arg name="base_frame"            default="$(arg robot_name)/base" />
    <arg name="camera_frame"          default="$(arg robot_name)/camera_frame" />
    <arg name="camera_pitch_radians"  default="0.35" /> <arg name="camera_pos_x"          default="0.080" />
    <arg name="camera_pos_y"          default="0.0" />
    <arg name="camera_pos_z"          default="0.058" />


    <node pkg="tf2_ros" type="static_transform_publisher" name="base_to_camera_frame_override"
          args="$(arg camera_pos_x) $(arg camera_pos_y) $(arg camera_pos_z) 0.0 $(arg camera_pitch_radians) 0.0 $(arg base_frame) $(arg camera_frame)" />


    <node pkg="White_detection" type="detection.py" name="line_detector_projector_avg" output="screen">
                                  <param name="target_frame" value="$(arg base_frame)" />

        <rosparam param="hsv_lower_white">[0, 0, 180]</rosparam>
        <rosparam param="hsv_upper_white">[180, 50, 255]</rosparam>

        <param name="roi_bottom_width_frac" value="0.9" type="double"/>
        <param name="roi_top_width_frac"    value="0.1" type="double"/>
        <param name="roi_height_frac"       value="0.4" type="double"/> <param name="blur_ksize"        value="5" type="int"/> <param name="canny_low_thresh"  value="50" type="int"/> <param name="canny_high_thresh" value="150" type="int"/> <param name="hough_rho"            value="1" type="int"/>
        <param name="hough_theta"          value="0.01745329" type="double"/> <param name="hough_threshold"      value="20" type="int"/> <param name="hough_min_line_len"   value="20" type="int"/> <param name="hough_max_line_gap"   value="5" type="int"/> <param name="min_line_slope_abs"   value="0.3" type="double"/>

    </node>

</launch>